modules/nextflow/src/main/groovy/nextflow/cloud/aws/batch/S3Helper.groovy

```diff
@@ -43,6 +43,40 @@ class S3Helper {
             unset IFS
         }
         
+        nxf_retry() {
+            local max_attempts=\${ATTEMPTS-4}
+            local timeout=\${TIMEOUT-1}
+            local attempt=0
+            local exitCode=0
+            echo "Downloading" \$2
+
+            while [[ \$attempt < \$max_attempts ]]
+            do
+              if "\$@"
+                then
+                  return 0
+              else
+                exitCode=\$?
+              fi
+              echo \$exitCode
+              if [[ \$exitCode == 0 ]]
+              then
+                echo "Download completed" \$2
```

Comment:
I would avoid debugging messages on the stdout, it can break task expected output. Let's only keep messages that report an error on the `stderr` channel.

Reply:
Removed all echos.